### PR TITLE
benchmark(hooks): dispatchable modules

### LINKED ISSUE #328: implement system hooks benchmarking tests
Take into account the hooks and the modules, which implement them.
* **Hooks**
  * [x] OnInitialize - @radkomih 
  * [x] OnFinalize - @radkomih 
  * [x] OnIdle - @radkomih 
  * [x] OnRuntimeUpgrade - @radkomih 
  ~~* [ ] OffchainWorker - @radkomih~~ 
   ~~* [ ] OnTimestampSet - @radkomih~~
  * [x] OnChargeTransaction @tgntr

### PR DESCRIPTION
**Pull request by radkomih**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #328 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -107,9 +107,5 @@ test-coverage:
...
 	./scripts/coverage.sh
 
-repeat=20
-steps=50
-heap-pages=4096
-db-cache=1024
 benchmark:
-	@GOMAXPROCS=1 go test --tags="nonwasmenv" -run=XXX -bench=. -benchtime=$(repeat)x ./runtime/... -steps=$(steps) -repeat=$(repeat) -heap-pages=$(heap-pages) -db-cache=$(db-cache);
\ No newline at end of file
+	@GOMAXPROCS=1 go test --tags="nonwasmenv" -run=XXX -bench=. ./runtime/... -steps=50 -repeat=20 -heap-pages=4096 -db-cache=1024
```
Comment (tgntr) [inline Makefile]: nit: is this on purpose?

**Review by tgntr — DISMISSED** [review]

File: Makefile (hunk 1 above)
Comment (radkomih) [reply Makefile]: yes, I don't see them used elsewhere

File: api/benchmarking/module.go [hunk 2]
```diff
@@ -39,31 +43,133 @@ func New(systemModule system.Module, decoder types.RuntimeDecoder, logger log.Lo
...
+	}
+	arg1, err := primitives.DecodeWeight(buffer)
+	if err != nil {
+		m.logger.Critical(err.Error())
+	}
+
+	benchmarkResult := m.executeBenchmark(config, func(origin primitives.RawOrigin) int64 {
+		var elapsed int64
+
+		// Benchmark the comulative time of dispatchable module hooks.
```
Comment (failfmi) [inline api/benchmarking/module.go]: nit: cumulative

File: api/benchmarking/module.go [hunk 3]
```diff
@@ -189,3 +277,11 @@ func (m Module) accountStorageKeyFrom(address primitives.AccountId) []byte {
...
 	return keyStorageAccount
 }
+
+func calculateAverageTime(durations []int64) int64 {
+	var sum int64
+	for _, duration := range durations {
+		sum += duration
+	}
+	return sum / int64(len(durations))
+}
```
Comment (failfmi) [inline api/benchmarking/module.go]: nit: use `stats` pkg?

**Review by failfmi — APPROVED** [review]

**Pull request merged**